### PR TITLE
Update Terraform-managed RDS instances - rds module

### PR DESCRIPTION
**Pull request by ccostino**
This changeset is part of a multi-step series of actions to upgrade our Terraform-managed RDS instances that are internal to the cloud.gov platform.  When all steps are completed we should see the following:

- All RDS instances running on the latest version available
- RDS instances down-sized to `db.t3.micro` instances where appropriate, or `db.t3.small`/`db.t3.medium` where still necessary

In order to accomplish this, the instances have to be upgraded in phases since you cannot go from a `9.6.x` release straight to a `12.x` release.  Furthermore, the `db.t3.*` instance class is not available for anything below PostgreSQL `11.5` (all internal databases are PostgreSQL instances).  In our other work updating and down-sizing other PostgreSQL instances, we have found that we can upgrade from a `9.6.x` release to `11.1`, then go to the latest `12.x` release (`12.3` at the time this PR was opened), provided there are no extra extensions enabled or `undefined` field types.

## Initial changes proposed in this pull request:
- Update all PostgreSQL instances to version `11.1`
- Update default parameter group to `default.postgres11`

## Future/upcoming proposed changes (separate steps/PRs):
1. Update all PostgreSQL instances to version `12.3`
1. Update PostgreSQL instance classes to `db.t3.micro` where appropriate; `db.t3.small`/`db.t3.medium` where necessary

## Security considerations
None:  these are modifications to existing cloud infrastructure resources defined in a publicly available repository that simply update versions and eventually change instance class designations.

## Maintenance considerations

This will incur some downtime as the instances are upgraded, even those that are set for Multi AZ.  We will need to coordinate when it will be best to go through these updates and changes, especially as we'll have to run through this multiple times to update to the latest version and then apply the instance class changes.

### CODE REVIEW

File: terraform/modules/rds/variables.tf [hunk 1]
```diff
@@ -45,7 +45,7 @@ variable "rds_parameter_group_name" {
 }
 
 variable "rds_parameter_group_family" {
-  default = "postgres9.6"
+  default = "default.postgres11"
```
Comment (ccostino) [inline terraform/modules/rds/variables.tf:48]: Is this even necessary? This was just referring to the default parameter group, can we just remove this all together and let Terraform and AWS go with the default themselves?

**Comment by jfredrickson5** [comment]
AWS RDS now allows upgrading from `9.6.19` to `12.4` directly. I updated the changes in this PR so that it now targets engine version `12.4` and parameter group family `postgres12` instead of `11.1` and `postgres11` respectively.

**Comment by bengerman13** [comment]
Since this one updates the default, it should probably be the last to be merged, right?

**Comment by ccostino** [comment]
I'm not sure of the proper order to merge all of these PRs, but if this is the case, yes on this one.  @ChrisMcGowan do you hav a sense of what order all of these should go in once they're ready?

**Comment by jfredrickson5** [comment]
We can close this PR since I ended up having to make this particular change to implement another RDS instance upgrade.

**Pull request closed without merging**